### PR TITLE
Metricbeat: Add mapping for docker metrics per cpu

### LINKED ISSUE #6785: Problem with mapping in metricbeat
This is related to https://github.com/elastic/beats/issues/5032

I think from reading the thread the issues is that we need to pass the number 0 as a float ?

Right now as-is a fresh install results in an error like this:

```
[2018-04-01T20:00:05,442][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"metricbeat-6.2.3-2018.04.02", :_type=>"doc", :_routing=>nil}, #<LogStash::Event:0x2aa897b5>], :response=>{"index"=>{"_index"=>"metricbeat-6.2.3-2018.04.02", "_type"=>"doc", "_id"=>"2_Wng2IBJsKPXoGhVOah", "status"=>400, "error"=>{"type"=>"illegal_argument_exception", "reason"=>"mapper [docker.cpu.core.23.pct] cannot be changed from type [long] to [float]"}}}}
```

### PR DESCRIPTION
**Pull request by jsoriano**
Fixes #6785

### CODE REVIEW

**Comment by ruflin** [comment]
This will probably not work. You have to use dynamic templates for this: https://www.elastic.co/guide/en/elasticsearch/reference/6.2/dynamic-templates.html

So far the way we use dynamic templates is that we specify it as following:

```
    - name: labels
      type: object
      object_type: keyword
      description: >
        Image labels.
```

In the template this outputs:

```
        {
          "docker.image.labels": {
            "mapping": {
              "type": "keyword"
            },
            "match_mapping_type": "string",
            "path_match": "docker.image.labels.*"
          }
        },
```

I could be that for this case an extension to our template generation is needed. Best have a look at `libbeat/template/processor.go` and create a test in `processor_test.go` to see if you get the expected output.

File: libbeat/template/processor.go [hunk 1]
```diff
@@ -208,6 +209,21 @@ func (p *Processor) object(f *common.Field) common.MapStr {
 	}
 
 	switch f.ObjectType {
+	case "scaled_float":
```
Comment (ruflin) [inline libbeat/template/processor.go]: Probably we will have a request for float and integer here pretty soon. As soon this pops up we should refactor the code to remove the duplicated parts, if possible

Comment (ruflin) [reply libbeat/template/processor.go]: Just saw further below that long we already have. float should be rather simple to add in the same manner.

**Comment by ruflin** [comment]
Change looks good to, looking forward to your manual testing. The template can be exported with `metricbeat export template`.

File: libbeat/template/processor.go (hunk 1 above)
Comment (jsoriano) [reply libbeat/template/processor.go]: For scaled float I could reuse the existing `scaledFloat` method.

File: libbeat/template/processor.go [hunk 2]
```diff
@@ -208,6 +209,9 @@ func (p *Processor) object(f *common.Field) common.MapStr {
...
+		dynProperties = p.scaledFloat(f)
+		addDynamicTemplate(f, dynProperties, matchType("*"))
```
Comment (jsoriano) [inline libbeat/template/processor.go]: I use `*` by default because we want any type to match at least all numeric types.

**Comment by jsoriano** [comment]
jenkins, test it again please, it seems that there has been an issue with pypi

**Comment by jsoriano** [comment]
jenkins, test it again please

**Comment by jsoriano** [comment]
jenkins, test it

**Comment by jsoriano** [comment]
jenkins, test this again

**Comment by jsoriano** [comment]
Yes, I could test it locally, it works fine :)

Branch rebased.

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -79,6 +79,8 @@ https://github.com/elastic/beats/compare/v6.0.0-beta2...master[Check the HEAD di
 - Fix Kubernetes calculated fields store. {pull}6564{6564}
 - Exclude bind mounts in fsstat and filesystem metricsets. {pull}6819[6819]
 - Don't stop Metricbeat if aerospike server is down. {pull}6874[6874]
+- Fix Kubernetes calculated fields store. {pull}6564[6564]
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: Seems like that line of changelog sneaked in?

Comment (jsoriano) [reply CHANGELOG.asciidoc]: Umm, I don't know why this happens :? removing it

Comment (ruflin) [reply CHANGELOG.asciidoc]: rebase with 3 way merges. Unfortunately a common issue.

**Comment by ctindel** [comment]
@ruflin Will this go into any of the 6.x releases as well?

**Pull request merged**